src/redfish/messages/messages.py

```diff
@@ -0,0 +1,174 @@
+        if "@Message.ExtendedInfo" in message_body:
+            messages_detail["@Message.ExtendedInfo"] = message_body["@Message.ExtendedInfo"]
+    except:
+        messages_detail["code"] = ""
+        messages_detail["message"] = ""
+        messages_detail["@Message.ExtendedInfo"] = []
+
+    return messages_detail
+
+def search_message(response, message_registry_group, message_registry_id):
+    """
+    search message in the payload
+
+    Args:
+        response: The response to parser
+        message_registry_group: target message_registry_group
+        message_registry_id: target message_registry_id
+    Returns:
+        The dict containing target message detail
+    """
+    if isinstance(response, dict) and "@Message.ExtendedInfo" in response:
+        messages_detail = response
+    else:
+        messages_detail = get_messages_detail(response)
+    message_registry_id_search = "^" + message_registry_group + "\.[0-9]+\.[0-9]+\." + message_registry_id +"$"
+
+    for messages_item in messages_detail["@Message.ExtendedInfo"]:
+        if "MessageId" in messages_item:
+            resault = re.search(message_registry_id_search, messages_item["MessageId"])
+            if resault:
+                if not "@odata.type" in messages_item:
+                    messages_item["@odata.type"] = ""
+                if not "RelatedProperties" in messages_item:
+                    messages_item["RelatedProperties"] = []
+                if not "Message" in messages_item:
+                    messages_item["Message"] = ""
+                if not "MessageArgs" in messages_item:
+                    messages_item["MessageArgs"] = []
+                if not "Severity" in messages_item:
+                    messages_item["Severity"] = ""
+                if not "MessageSeverity" in messages_item:
+                    messages_item["MessageSeverity"] = ""
+                if not "Resolution" in messages_item:
+                    messages_item["Resolution"] = ""
+                return messages_item
+    return None
+
+def get_error_messages( response ):
+    """
+    Builds a string based on the error messages in the payload
+
+    Args:
+        response: The response to print
+
+    Returns:
+        The string containing error messages
+    """
+
+    # Pull out the error payload and the messages
+
+    out_string = ""
+    try:
+        out_string = response.dict["error"]["message"]
+        if "@Message.ExtendedInfo" in response.dict["error"]:
+            for message in response.dict["error"]["@Message.ExtendedInfo"]:
+                if "Message" in message:
+                    out_string = out_string + "\n" + message["Message"]
+            else:
+                    out_string = out_string + "\n" + message["MessageId"]
+        out_string = out_string + "\n"
+    except:
+        # No response body
+        out_string = ""
+
+    return out_string
+
+def verify_response( response ):
```

Comment:
I think we can omit this for now; it seems like it might be good to add in the future to try to have consuming applications leverage this, but I think we need to consider how to layer this on top of other error handling done at the application-level. I can see it potentially causing conflicts with exception handling on consumers of the library, so we'd need to be careful about adding this in when we think it's a good time to do so.

Reply:
according to feedback, should I move verify_response back to Redfish-Tacklebox

Reply:
Yes, for now lets keep the verify function at the Tacklebox layer.